component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java

```diff
@@ -855,6 +855,65 @@ public Response deleteLike(
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "500", description = "Internal server error"),
+      @ApiResponse(responseCode = "400", description = "Invalid query input") })
+  public Response pinActivity(@Context UriInfo uriInfo,
+                              @Context Request request,
+                              @Parameter(description = "Activity id", required = true) @PathParam("activityId") String activityId) {
+
+    if (StringUtils.isBlank(activityId)) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    org.exoplatform.services.security.Identity authenticatedUserIdentity = ConversationState.getCurrent().getIdentity();
+    String authenticatedUser = authenticatedUserIdentity.getUserId();
+    Identity currentUser = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, authenticatedUser);
+
+    ExoSocialActivity activity = activityManager.getActivity(activityId);
+
+    if (!activityManager.canPinActivity(activity, currentUser)) {
+      return Response.status(Status.UNAUTHORIZED).build();
+    }
+    ActivityEntity activityEntity;
+    activity = activityManager.pinActivity(activity.getId(), currentUser.getId());
+    activityEntity = EntityBuilder.buildEntityFromActivity(activity, currentUser, uriInfo.getPath(), null);
+    return EntityBuilder.getResponse(activityEntity.getDataEntity(), uriInfo, RestUtils.getJsonMediaType(), Response.Status.OK);
+  }
+
+  @DELETE
+  @Path("{activityId}/pins")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @Operation(summary = "Unpin a specific activity from space stream", method = "DELETE", description = "This Unpins an activity from space stream")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "500", description = "Internal server error"),
+      @ApiResponse(responseCode = "400", description = "Invalid query input") })
+  public Response unpinActivity(@Context UriInfo uriInfo,
+                                @Context Request request,
+                                @Parameter(description = "Activity id", required = true) @PathParam("activityId") String activityId) {
+
+    if (StringUtils.isBlank(activityId)) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    org.exoplatform.services.security.Identity authenticatedUserIdentity = ConversationState.getCurrent().getIdentity();
+    String authenticatedUser = authenticatedUserIdentity.getUserId();
+    Identity currentUser = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, authenticatedUser);
+
+    ExoSocialActivity activity = activityManager.getActivity(activityId);
+
+    if (!activityManager.canPinActivity(activity, currentUser)) {
```

Comment:
What if activity is null (activity not founded) ?

Reply:
It seems to generate a NPE.

Reply:
Done!